src/widgets/helper/ResizingTextEdit.cpp

```diff
@@ -264,14 +265,32 @@ bool ResizingTextEdit::canInsertFromMimeData(const QMimeData *source) const
 
 void ResizingTextEdit::insertFromMimeData(const QMimeData *source)
 {
-    if (source->hasImage() || source->hasUrls())
+    if (source->hasImage())
     {
         this->imagePasted.invoke(source);
+        return;
     }
-    else
+    else if (source->hasUrls())
     {
-        insertPlainText(source->text());
+        bool hasUploadable = false;
+        auto mimeDb = QMimeDatabase();
+        for (const QUrl url : source->urls())
+        {
+            QMimeType mime = mimeDb.mimeTypeForUrl(url);
+            if (mime.name().startsWith("image"))
+            {
+                hasUploadable = true;
```

Comment:
Is the logic "If the list of urls contains at minimum one image url, upload it?" - if so, we could early out here.
```suggestion
                hasUploadable = true;
                break;
```